bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/codemining/LineContentCodeMining.java

```diff
@@ -44,7 +62,25 @@ public LineContentCodeMining(Position position, ICodeMiningProvider provider) {
 	 * @param action the action to execute when mining is clicked and null otherwise.
 	 */
 	public LineContentCodeMining(Position position, ICodeMiningProvider provider, Consumer<MouseEvent> action) {
+		this(position, false, provider, action);
+	}
+
+	/**
+	 * CodeMining constructor to locate the code mining in a given position.
+	 *
+	 * @param position the position where the mining must be drawn.
+	 * @param provider the owner codemining provider which creates this mining.
+	 * @param action the action to execute when mining is clicked and null otherwise.
+	 * @param afterPosition if true code mining is treated as suffix code mining where cursor and
+	 *            selection is not including the mining
+	 */
+	public LineContentCodeMining(Position position, boolean afterPosition, ICodeMiningProvider provider, Consumer<MouseEvent> action) {
```

Comment:
Since this is new API, this will likely require some `@since 3.26` and a version increase in the MANIFEST.